wsd/DocumentBroker.cpp

```diff
@@ -3723,6 +3765,66 @@ bool DocumentBroker::isAsyncUploading() const
     return state == StorageBase::AsyncUpload::State::Running;
 }
 
+std::string DocumentBroker::addEmbeddedMedia(const std::string& json)
+{
+    Poco::JSON::Object::Ptr object;
+    if (JsonUtil::parseJSON(json, object))
+    {
+        const std::string id = JsonUtil::getJSONValue<std::string>(object, "id");
+        if (id.empty())
+        {
+            LOG_ERR("Invalid embeddedmedia json without id: " << json);
+        }
+        else
+        {
+            LOG_TRC("Adding embeddedmedia with id [" << id << "]: " << json);
+
+            // Store the original json with the internal, temporary, file URI.
+            _embeddedMedia[id] = json;
+
+            std::string wopiSrc;
+            Poco::URI::encode(Util::split(_uriPublic.toString(), '?').first, ":?#/=&", wopiSrc);
+
+            Poco::URI uri("/cool/media");
+            uri.setScheme(_uriPublic.getScheme());
+            uri.setHost(_uriPublic.getHost());
+            uri.setPort(COOLWSD::getClientPortNumber());
+            uri.addQueryParameter("ServerId", Util::getProcessIdentifier());
+            uri.addQueryParameter("Tag", id);
+            uri.addQueryParameter("WOPISrc", wopiSrc);
+
+            std::string mediaUrl;
+            Poco::URI::encode(uri.toString(), "&", mediaUrl); // '&' is reserved in xml/html/svg.
+            object->set("url", mediaUrl);
+            object->set("mimeType", "video/ogg");
```

Comment:
Here we have `video/ogg` but above in `DocumentBroker::handleMediaRequest` we always put `video/mp4`
Is that correct?

Reply:
Fixed (although I don't think it makes a difference in practice, and anyway we don't know the actual media mime, but best to be consistent, of course).